### DISCUSSION TITLE
I want to disable backspace for any  shape delete

### DISCUSSION THREAD

**Discussion by robertwilli1** [pikasojs/pikaso#52] [Q&A] [answered]
I tried the following code to disable the shapes being deleted on the press of backspace key  :
selection: {
      interactive:true,
      keyboard:{   
         enabled: true,
         movingSpaces:10,
          map: {
            delete: ['Delete'],
          }
        },

but this is not working 
Can you please tell me how I can disable backspace key so that my shapes do not get deleted on press of backspace key.

**Comment by raminious** [comment, marked as answer]
Hi @osamatan125 

Try this please


```ts
const pikaso = new Pikaso({
  container: document.getElementById("app")!,
  selection: {
    keyboard: {
      enabled: false,
      map: {
        delete: []
      }
    }
  }
})

```

https://codesandbox.io/s/restless-waterfall-nxz76o?file=/src/index.ts

**Reply by robertwilli1** [reply]
But sir I want to enable keyboard keys, I just want that shapes should not be deleted by Backspace key and should be deleted by Delete key.
Is it possible?

**Reply by raminious** [reply]
Yes, did you check the codesandbox I've shared with you?

**Reply by robertwilli1** [reply]
Yes, I have  checked the code sandbox example and added the code you sent me in the previous reply but when I set keyboard: { enabled: false} it does not delete any  shape from any key on the keyboard and when I set keyboard enabled: true it deletes shapes from "delete" and "backspace" key.
What I want is that shapes should only be deleted by "delete" key and not by "backspace" key.

**Reply by raminious** [reply]
I think this is not possible. for example on Macbook keyboards there is one key for Backspace and Delete.

**Reply by robertwilli1** [reply]
ok.
